types/list.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	"sync"
+)
+
+type List struct {
```

Comment:
Wasn't the point of #387 to phase out these structures?

Reply:
Originally yes, but I found myself building something very similar and thought that they were the best solution for now